modules/sc-mesh-secure-deployment/src/nats/src/cbma_adaptation.py

```diff
@@ -646,10 +647,7 @@ def setup_cbma(self) -> bool:
         # Set batman hop penalty
         self.__batman.set_hop_penalty()
 
-        self.__cbma_set_up = True
-
-        # todo what about the false return value?
-        return True
+        return lower_cbma_set_up and upper_cbma_set_up
```

Comment:
I believe we should `return True` here and early `return False` if setting up lower or upper-CBMA fails:
```python
if not self.__setup_lower_cbma() or not self.__setup_upper_cbma():
    return False
```
:trackball:

Reply:
True, it would make more sense but to be honest error handling would still require more work. As I run into SECO-5165 I run into a problem that upper CBMA setup started to fail because new certificates are invalid. Now with this implementation it causes setup_cbma() to fail which again leads to a situation that MDM agent tries again after its fail timeout (1s) which then leads yet to another problems because for the new round we still have for example lmb interfaces created in previous lower cbma setup round which leads to a situation that lmb interfaces are treated as black interfaces thus they are tried to be added to lower cbma which causes problems and subseguent trial to setup upper cbma would still fail due to invalid certificates. I was thinking to improve those things whilst doing unit tests for CBMA Adaptation. However, I don't know yet how I should deal cases where received upper CBMA certificates are invalid. Should I juts try to revert to birth certs? I created a seprate ticket for further improvents. SECO-5182.

Reply:
setup_cbma() updated as Carlos suggest with one addition to call stop_cbma() in case of failure. Also stop_cbma() itself was refactored a bit to use separate helper method to stop cbma.

Reply:
Hmmm, I believe the `lmb` interface should be deleted if the associated CBMA instance is stopped (unless there's something else attached to the bridge not added by CBMA when it stops), so the scenario you described shouldn't happen if all CBMA instances are stopped if this method returns `False` before trying again...
Alternatively, maybe the `lmb` interfaces should be added to the `__na_cbma_interfaces` list (and it shall support wildcards) :thinking:

Reply:
Ah, stopping all CBMA instances in case of failure is what you did in [48c4ea9](https://github.com/tiiuae/mesh_com/pull/451/commits/48c4ea964d9ab59e16213b7e62dc0d5983a798b0), awesome! :star2: